lamagraph-compiler/test/Lamagraph/Compiler/TokenPassingCBVGolden.hs

```diff
@@ -0,0 +1,143 @@
+module Lamagraph.Compiler.TokenPassingCBVGolden (
+  tokenPassingCBVLmlGolden,
+  tokenPassingCBVParallelLmlGolden,
+  tokenPassingCBVCoreGolden,
+  tokenPassingCBVParallelCoreGolden,
+) where
+
+import Relude
+
+import Data.Either.Extra (mapLeft)
+import Data.Foldable
+import Data.HashMap.Strict qualified as HashMap
+import Prettyprinter hiding (width)
+import Prettyprinter.Render.Text
+import System.FilePath
+import Test.Tasty
+import Test.Tasty.Golden
+import UnliftIO.Exception
+
+import Lamagraph.Compiler.Core
+import Lamagraph.Compiler.Core.LmlToCore
+import Lamagraph.Compiler.Core.MonadDesugar
+import Lamagraph.Compiler.Core.Pretty ()
+import Lamagraph.Compiler.GoldenCommon
+import Lamagraph.Compiler.Nets.Encodings.LambdaCalculusCore
+import Lamagraph.Compiler.Nets.Encodings.TokenPassingCBV
+import Lamagraph.Compiler.Nets.Processing
+import Lamagraph.Compiler.Nets.Reduction
+import Lamagraph.Compiler.Nets.Types
+import Lamagraph.Compiler.Parser
+import Lamagraph.Compiler.Typechecker.Infer
+
+newExt :: String
+newExt = "out"
+
+lmlInputDir :: FilePath
+lmlInputDir = baseGoldenTestsDir </> "nets" </> "token_passing_cbv_lml_in"
+
+lmlOutputDir :: FilePath
+lmlOutputDir = ".." </> "token_passing_cbv_lml_out"
+
+coreOutputDir :: FilePath
+coreOutputDir = baseGoldenTestsDir </> "nets" </> "token_passing_cbv_core_out"
+
+renderUnbounded :: (Pretty a) => a -> LByteString
+renderUnbounded x = encodeUtf8 $ renderLazy $ layoutPretty (LayoutOptions{layoutPageWidth = Unbounded}) (pretty x)
+
+analyzeStats ::
+  INsMachineStats ->
+  -- | Reduction count, reduction width history, parallel width, parallel height
+  ( Word64
```

Comment:
Can result of analysis be declared as a type (record?) to have named field and to avoid explanation comments?